ftplugin/c.vim

```diff
@@ -1,6 +1,8 @@
 let s:cb = {->
-	 \ switchy#has_suffix(bufname(''), '.c')
-         \ ? bufname('')[:-3] .. '.h'
```

Comment:
Is there a reason why `bufname('')[:-3]` is preferred to `expand('%:r')`? Using `%` seems more robust as I am not sure that the buffer name always has to coincide with its underlying file path? Another advantage would be that the indices don't need to be computed depending on the length of the extension, which could lead to copy-and-paste oversights.

Reply:
As far as I know bufname('') should always be the filename, at least for buffers with a file.

I used bufname('') when I was working vim-go, which is used by a lot of people, and never had any complaints about that.

expand('%:r') should also be fine, I think(?) I don't recall why I choose to use bufname('') back in the day. Probably just because it's easier to type and read.

Anyway, I made it more readable in https://github.com/arp242/switchy.vim/commit/bc9e35e845febec637f9d34738e1456c5f364f26